pkg/policy/consumer.go

```diff
@@ -253,11 +215,27 @@ func (c *Consumable) RemoveIngressIdentityLocked(id NumericIdentity) {
 		log.WithField(logfields.Identity, id).Debug("Removing identity from ingress map")
 		delete(c.IngressIdentities, id)
 
+		// Consumables corresponding to reserved identities need to be updated
+		// explicitly because they are not updated or regenerated.
+		if id.IsReservedIdentity() {
+			reservedConsumable := c.cache.Lookup(id)
+			if reservedConsumable != nil {
+				// Avoid deadlock!
+				if id != c.ID {
+					reservedConsumable.Mutex.Lock()
+					reservedConsumable.RemoveIngressIdentityLocked(c.ID)
+					reservedConsumable.Mutex.Unlock()
+				} else {
+					reservedConsumable.RemoveIngressIdentityLocked(c.ID)
```

Comment:
same here?

Reply:
Same as above reply in `AllowIngressIdentityLocked`.